### ISSUE TITLE
VO xv segfaults when 4K is played

### ISSUE THREAD

**Issue by Nikoli** [mpv-player/mpv#320]
For one of my systems playing 4K files causes mov segfault when vo xv is used:

```
$ mpv -V
mpv git-74e7043 (C) 2000-2013 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
libav library versions:
   libavutil       52.3.0
   libavcodec      54.35.0
   libavformat     54.20.3
   libswscale      2.1.1
   libavfilter     3.3.0
   libavresample   1.0.1
```

xf86-video-nouveau-1.0.9, mesa-9.1.6

bt:

```
Starting program: /usr/bin/mpv Sintel.2010.4K.DMRip.Hi444PP.x264.DTSMA.SRT-MaLLIeHbKa.mkv -vo xv
warning: Cannot call inferior functions, Linux kernel PaX protection forbids return to non-executable pages!
warning: Could not load shared library symbols for linux-vdso.so.1.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
[New Thread 0x3a792d74700 (LWP 4171)]
[New Thread 0x3a79255a700 (LWP 4172)]

Program received signal SIGSEGV, Segmentation fault.
0x000003a79cd43c1b in ?? () from /lib64/libc.so.6
#0  0x000003a79cd43c1b in ?? () from /lib64/libc.so.6
No symbol table info available.
#1  0x0000005d0b646eb6 in memset (__len=2048, __ch=128, __dest=<optimized out>) at /usr/include/bits/string3.h:85
No locals.
#2  memset_pic (stride=1024, height=872, bytesPerLine=2048, fill=128, dst=<optimized out>) at video/memcpy_pic.h:67
        i = <optimized out>
#3  mp_image_clear (img=img@entry=0x3e3470a3c20, x0=x0@entry=0, y0=y0@entry=0, x1=x1@entry=4096, y1=<optimized out>) at video/mp_image.c:416
        bpp = <optimized out>
        bytes = 2048
        p = <optimized out>
        __PRETTY_FUNCTION__ = "mp_image_clear"
        area = {flags = 29441, fmt = {id = 1004, avformat = 0, name = 0x5d0b6a4ec4 "420p", flags = 29441, num_planes = 3 '\003', chroma_xs = 1 '\001', chroma_ys = 1 '\001', 
            align_x = 2 '\002', align_y = 2 '\002', bytes = "\001\001\001", bpp = "\b\b\b", plane_bits = 8 '\b', xs = "\000\001\001", ys = "\000\001\001"}, 
          imgfmt = IMGFMT_420P, num_planes = 3, chroma_x_shift = 1, chroma_y_shift = 1, w = 4096, h = 1744, display_w = 4096, display_h = 1744, planes = {0x3a790bf0000 "", 
            0x3a791032000 "\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200"..., 0x3a790f58000 "", 0x0}, stride = {2048, 1024, 
            1024, 0}, qscale = 0x0, qstride = 0, pict_type = 0, fields = 0, qscale_type = 0, chroma_width = 2048, chroma_height = 872, plane_w = {4096, 2048, 2048, 0}, 
          plane_h = {1744, 872, 872, 0}, colorspace = MP_CSP_BT_709, levels = MP_CSP_LEVELS_TV, chroma_location = MP_CHROMA_AUTO, pts = 0, refcount = 0x0, priv = 0x0}
        plane_clear = {0, 128, 128, 0}
#4  0x0000005d0b6907eb in allocate_xvimage (foo=0, vo=0x5d0bcef9f0) at video/out/vo_xv.c:528
        ctx = 0x5d0bce0640
        x11 = 0x5d0bcf46a0
        aligned_w = 4096
        img = {flags = 29441, fmt = {id = 1004, avformat = 0, name = 0x5d0b6a4ec4 "420p", flags = 29441, num_planes = 3 '\003', chroma_xs = 1 '\001', chroma_ys = 1 '\001', 
            align_x = 2 '\002', align_y = 2 '\002', bytes = "\001\001\001", bpp = "\b\b\b", plane_bits = 8 '\b', xs = "\000\001\001", ys = "\000\001\001"}, 
          imgfmt = IMGFMT_420P, num_planes = 3, chroma_x_shift = 1, chroma_y_shift = 1, w = 4096, h = 1744, display_w = 4096, display_h = 1744, planes = {0x3a790bf0000 "", 
            0x3a791032000 "\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200\200"..., 0x3a790f58000 "", 0x0}, stride = {2048, 1024, 
            1024, 0}, qscale = 0x0, qstride = 0, pict_type = 0, fields = 0, qscale_type = 0, chroma_width = 2048, chroma_height = 872, plane_w = {4096, 2048, 2048, 0}, 
          plane_h = {1744, 872, 872, 0}, colorspace = MP_CSP_BT_709, levels = MP_CSP_LEVELS_TV, chroma_location = MP_CHROMA_AUTO, pts = 0, refcount = 0x0, priv = 0x0}
#5  config (vo=0x5d0bcef9f0, width=<optimized out>, height=<optimized out>, d_width=<optimized out>, d_height=<optimized out>, flags=0, format=1004) at video/out/vo_xv.c:472
        x11 = 0x5d0bcf46a0
        ctx = 0x5d0bce0640
        i = <optimized out>
#6  0x0000005d0b660127 in vo_reconfig (vo=0x5d0bcef9f0, params=0x3e3470a3e10, flags=0) at video/out/vo.c:419
        d_width = 4096
        d_height = 1744
        p2 = {imgfmt = IMGFMT_420P, w = 4096, h = 1744, d_w = 4096, d_h = 1744, colorspace = MP_CSP_BT_709, colorlevels = MP_CSP_LEVELS_TV, chroma_location = MP_CHROMA_LEFT, 
          outputlevels = MP_CSP_LEVELS_AUTO}
        ret = <optimized out>
#7  0x0000005d0b64c06d in vf_reconfig_wrapper (vf=0x5d0bd0d790, p=0x3e3470a3e90, flags=0) at video/filter/vf.c:455
        params = {imgfmt = IMGFMT_420P, w = 4096, h = 1744, d_w = 4096, d_h = 1744, colorspace = MP_CSP_BT_709, colorlevels = MP_CSP_LEVELS_TV, 
          chroma_location = MP_CHROMA_LEFT, outputlevels = MP_CSP_LEVELS_AUTO}
        r = <optimized out>
#8  0x0000005d0b64c06d in vf_reconfig_wrapper (vf=0x5d0c620780, p=p@entry=0x3e3470a3f30, flags=flags@entry=0) at video/filter/vf.c:455
        params = {imgfmt = IMGFMT_420P, w = 4096, h = 1744, d_w = 4096, d_h = 1744, colorspace = MP_CSP_BT_709, colorlevels = MP_CSP_LEVELS_TV, 
          chroma_location = MP_CHROMA_LEFT, outputlevels = MP_CSP_LEVELS_AUTO}
        r = <optimized out>
#9  0x0000005d0b64954a in mpcodecs_reconfig_vo (sh=sh@entry=0x5d0ba89f40, params=params@entry=0x3e3470a3fe0) at video/decode/vd.c:148
        opts = 0x5d0ba434e0
        vf = 0x5d0c620780
        vocfg_flags = 0
        p = {imgfmt = IMGFMT_444P10_LE, w = 4096, h = 1744, d_w = 4096, d_h = 1744, colorspace = MP_CSP_BT_709, colorlevels = MP_CSP_LEVELS_TV, 
          chroma_location = MP_CHROMA_LEFT, outputlevels = MP_CSP_LEVELS_AUTO}
        flags = <optimized out>
        flip = false
        force_aspect = <optimized out>
#10 0x0000005d0b64aa15 in decode (sh=sh@entry=0x5d0ba89f40, packet=packet@entry=0x5d0c1a90f0, flags=flags@entry=0, reordered_pts=reordered_pts@entry=0x3e3470a4218, 
    out_image=out_image@entry=0x3e3470a41b0) at video/decode/vd_lavc.c:754
        got_picture = 1
        ret = <optimized out>
        ctx = 0x5d0c085a00
        pic = 0x5d0c085ae0
        avctx = <optimized out>
        pkt = {pts = -9223372036854775808, dts = -9223372036854775808, data = 0x5d0c18aad0 "", size = 206, stream_index = 0, flags = 0, side_data = 0x0, side_data_elems = 0, 
          duration = 0, destruct = 0x0, priv = 0x0, pos = -1, convergence_duration = 0}
        mpi = 0x5d0c1a84a0
        __PRETTY_FUNCTION__ = "decode"
        vo_params = {imgfmt = IMGFMT_444P10_LE, w = 4096, h = 1744, d_w = 4096, d_h = 1744, colorspace = MP_CSP_BT_709, colorlevels = MP_CSP_LEVELS_TV, 
          chroma_location = MP_CHROMA_LEFT, outputlevels = MP_CSP_LEVELS_AUTO}
#11 0x0000005d0b64ab28 in decode_with_fallback (sh=0x5d0ba89f40, packet=0x5d0c1a90f0, flags=0, reordered_pts=0x3e3470a4218) at video/decode/vd_lavc.c:774
        ctx = 0x5d0c085a00
        mpi = 0x0
        res = <optimized out>
#12 0x0000005d0b649000 in decode_video (sh_video=sh_video@entry=0x5d0ba89f40, packet=0x5d0c1a90f0, drop_frame=0, pts=0) at video/decode/dec_video.c:257
        mpi = 0x0
        opts = 0x5d0ba434e0
        prevpts = <optimized out>
#13 0x0000005d0b623b58 in update_video (endpts=-9.2233720368547758e+18, mpctx=0x5d0ba42bc0) at mpvcore/mplayer.c:2735
        pkt = <optimized out>
        framedrop_type = <optimized out>
        decoded_frame = <optimized out>
        sh_video = 0x5d0ba89f40
        video_out = 0x5d0bcef9f0
        pts = <optimized out>
        frame_time = <optimized out>
#14 run_playloop (mpctx=mpctx@entry=0x5d0ba42bc0) at mpvcore/mplayer.c:3675
        frame_time = 0
        t2 = <optimized out>
        pts_us = <optimized out>
        duration = <optimized out>
        vo = 0x5d0bcef9f0
        vsleep = <optimized out>
        sh_video = <optimized out>
        time_frame = 0
        pts2 = <optimized out>
        opts = 0x5d0ba434e0
        full_audio_buffers = false
        audio_left = false
        video_left = false
        endpts = -9.2233720368547758e+18
        end_is_chapter = false
        sleeptime = <optimized out>
        was_restart = true
        new_frame_shown = false
        buffered_audio = -1
#15 0x0000005d0b5dea94 in play_current_file (mpctx=<optimized out>) at mpvcore/mplayer.c:4656
        opts = 0x5d0ba434e0
        res = <optimized out>
        startpos = <optimized out>
        uninitialize_parts = <optimized out>
        playback_short = <optimized out>
        init_failed = <optimized out>
        playback_start = 10.080209999999999
        stream_filename = <optimized out>
#16 play_files (mpctx=<optimized out>) at mpvcore/mplayer.c:4780
        new_entry = <optimized out>
#17 mpv_main (argv=<optimized out>, argc=<optimized out>) at mpvcore/mplayer.c:5050
        mpctx = <optimized out>
        opts = <optimized out>
        r = <optimized out>
#18 main (argc=<optimized out>, argv=<optimized out>) at mpvcore/mplayer.c:5062
No locals.
A debugging session is active.

        Inferior 1 [process 4167] will be killed.

Quit anyway? (y or n)
```

**Comment by zaza42** [comment]
Ehm. mpv had worked perfectly before this checking. It always miscalculates the correct size by 1 pixel with my buggy sis671 card. I had to remove this code to watch movies again. :-/

```
[vo/xv] Got XvImage with incorrect size: 1280x772 (expected 1280x771)
[vo/xv] could not allocate Xv image data
Could not initialize video chain.
```

```
[vo/xv] Got XvImage with incorrect size: 640x386 (expected 640x385)
[vo/xv] could not allocate Xv image data
Could not initialize video chain.
```

**Comment by wm4** [comment]
Fixed via 22c76e85db88a772e3360892cd3a673a89c6fc7a.
